api/v1alpha1/randomsecret_types.go

```diff
@@ -83,6 +83,9 @@ func (d *RandomSecret) GetVaultConnection() *vaultutils.VaultConnection {
 }
 
 func (d *RandomSecret) GetPath() string {
+	if d.Spec.IsKVSecretsEngineV2 {
```

Comment:
we should check if the path already has "/data" in the prefix of the path and only add it if its missing for backwards compatibility. I'm actually surprised the integration test worked.